.github/workflows/mingw.yml

```diff
@@ -221,27 +221,21 @@ jobs:
 
 
   cross-llvm-mingw:
-    runs-on: ubuntu-18.04
+    runs-on: ubuntu-latest
     container:
       image: mstorsjo/llvm-mingw:latest
     strategy:
       matrix:
         arch: ['x86_64', 'i686', 'aarch64', 'armv7']
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
+
       - name: Install deps
         run: |
+          export DEBIAN_FRONTEND=noninteractive
+          add-apt-repository --yes ppa:deadsnakes/ppa
           apt-get update -qq
-          apt-get install -qqy autoconf-archive
-
-      - uses: actions/setup-python@v2
-        with:
-          python-version: '3.10'
-
-      - name: Check Python Version
-        run: |
-          which python
-          python --version
+          apt-get install -qqy autoconf-archive python3.10-dev python3.10
```

Comment:
I think you need to bump these two to 3.11 so it matches

Reply:
will do. also should I squash all the commits related to CI into a single one?

Reply:
Whatever is easier for you

Reply:
done, squashed all the CI commits into a single one -> 1c6cf39c377b9a66565e3aac7ab06127a8d87aaf